jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/ActionPrefixesR.java

```diff
@@ -0,0 +1,156 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.fuseki.servlets;
+
+import com.google.gson.JsonArray;
+import org.apache.jena.atlas.logging.FmtLog;
+import org.apache.jena.riot.WebContent;
+import org.apache.jena.riot.web.HttpNames;
+
+import org.apache.jena.fuseki.servlets.prefixes.ActionPrefixesBase;
+import org.apache.jena.fuseki.servlets.prefixes.PrefixUtils;
+import org.apache.jena.fuseki.servlets.prefixes.JsonObject;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class ActionPrefixesR extends ActionPrefixesBase {
+
+    private static final String NO_PREFIX_NS = "";
+
+    public ActionPrefixesR() {}
+
+    @Override
+    protected void doOptions(HttpAction action) {
+        ActionLib.setCommonHeadersForOptions(action);
+        action.setResponseHeader(HttpNames.hAllow, "GET,OPTIONS");
+        ServletOps.success(action);
+    }
+
+    public void validateGet(HttpAction action) {
+        validate(action);
+        // check if the combination of parameters is legal
+        String prefix = action.getRequestParameter("prefix");
+        String uri = action.getRequestParameter("uri");
+        if(prefix != null && uri != null) {
+            ServletOps.errorBadRequest("Provide only one of the prefix or uri parameters!");
+            return;
+        }
+    }
+
+    @Override
+    protected void doGet(HttpAction action) {
+        ActionLib.setCommonHeaders(action);
+        validateGet(action);
+
+        action.beginRead();
+        try {
+            // Not null (valid request)
+            String prefix = action.getRequestParameter("prefix");
+            String uri = action.getRequestParameter("uri");
+
+            if (prefix == null && uri == null) {
+                //getAll
+                Map<String, String> allPairs = prefixes(action).getAll();
+                JsonArray allJsonPairs = new JsonArray();
+                allPairs.entrySet().stream()
+                        .forEach(entry -> {
+                            com.google.gson.JsonObject jsonObject = new com.google.gson.JsonObject();
+                            jsonObject.addProperty("prefix", entry.getKey());
+                            jsonObject.addProperty("uri", entry.getValue());
+                            allJsonPairs.add(jsonObject);
+                            FmtLog.info(action.log, "[%d] - %s", action.id, new JsonObject(entry.getKey(), entry.getValue()));
+                        });
+                action.setResponseContentType(WebContent.contentTypeJSON);
+                action.getResponseOutputStream().print(String.valueOf(allJsonPairs));
+
+                ServletOps.success(action);
+                action.endRead();
+                return;
+            }
+            if (prefix != null && uri == null) {
```

Comment:
Stylistic (and entirely optional): Rather than these if checks which aren't particularly clear why not have a `chooseResponseType()` (or similarly named method) that does those in a short method returning an `enum` value indicating the response type selected and then using a `switch` statement here for clarity?